### PR TITLE
Activity feed

### PR DESCRIPTION
**Pull request by StevenAb**
Here is the bulk of the MVP for the activity feed. The page currently displays all conversations, messages, and registrations. The date stamps still need to be converted into a human readable format and sorted chronologically.

### CODE REVIEW

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 1]
```diff
@@ -33,13 +46,25 @@ void setUserStore(UserStore userStore) {
         this.userStore = userStore;
     }
     
+    //Sets the ConversationStore used by this servlet.
```
Comment (erschwartz) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: Please use docstring style comments for methods.

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 2]
```diff
@@ -33,13 +46,25 @@ void setUserStore(UserStore userStore) {
...
+    void setConversationStore(ConversationStore conversationStore) {
+        this.conversationStore = conversationStore;
+    }
+    
+    //Sets the MessageStore used by this servlet.
```
Comment (erschwartz) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: Docstring

File: src/main/java/codeu/model/store/basic/UserStore.java [hunk 3]
```diff
@@ -126,5 +126,16 @@ public boolean isUserRegistered(String username) {
   public void setUsers(List<User> users) {
     this.users = users;
   }
+
+  //Returns a list of all user IDs
```
Comment (erschwartz) [inline src/main/java/codeu/model/store/basic/UserStore.java]: Docstring

File: src/main/java/codeu/model/store/basic/UserStore.java [hunk 4]
```diff
@@ -126,5 +126,16 @@ public boolean isUserRegistered(String username) {
...
+  public List<UUID> getUserIds() {
+    List<UUID> userIds = new ArrayList();
+    
+    for(User user : users) {
+      userIds.add(user.getId());
+    }
+      
+    return userIds;
```
Comment (erschwartz) [inline src/main/java/codeu/model/store/basic/UserStore.java]: You can use Java 8 to do this in 1 line:

```java
return users.stream().map(User::getId).collect(Collectors.toList());
```

File: src/main/webapp/WEB-INF/view/activity-feed.jsp [hunk 5]
```diff
@@ -29,6 +37,62 @@
 
         <h1>Welcome to the activity feed.</h1>
 
+        <div id = "container">
+        <ul>
+        <%
+        List<Conversation> conversations =
+          (List<Conversation>) request.getAttribute("conversations");
+
+        if(conversations != null && !conversations.isEmpty()) {
```
Comment (erschwartz) [inline src/main/webapp/WEB-INF/view/activity-feed.jsp]: You don't need to check if it's empty as if it is the for loop will simply never execute

File: src/main/webapp/WEB-INF/view/activity-feed.jsp [hunk 6]
```diff
@@ -29,6 +37,62 @@
...
+                    conversation.getOwnerId()).getName() %> created conversation
+                  <a href="/chat/<%= conversation.getTitle() %>">
+                    <%= conversation.getTitle() %>
+                  </a>
+                </li>
+        <%
+                List<Message> messages = (List<Message>) MessageStore.getInstance()
+                  .getMessagesInConversation(conversation.getId());
+
+                if(messages != null && !messages.isEmpty()) {
```
Comment (erschwartz) [inline src/main/webapp/WEB-INF/view/activity-feed.jsp]: Same thing here, don't need to check if empty if it's part of for loop

**Review by erschwartz — CHANGES_REQUESTED** [review]
Nice work -- Can you add unit tests for this please?

**Comment by erschwartz** [comment]
Recapping what we discussed in the meeting:

You'll want to make each of the classes you wish to sort implement Comparable.
Then you'll want to have a method like:
```java
@Override
public int compareTo(Conversation c) {
        return this.creationTime.compareTo(c.creationTime);
}
```

Then in the jsp or perhaps in the servlet (up to you where you think it would be better to do this) you can do
```java
Collections.sort(conversations);
```

**Review by MatthewOakley — APPROVED** [review]
Besides what was stated everything LGTM!

File: src/main/java/codeu/model/store/basic/UserStore.java [hunk 7]
```diff
@@ -126,5 +126,18 @@ public boolean isUserRegistered(String username) {
...
     this.users = users;
   }
+
+  /**
+   *Returns a list of all user IDs
+   */
+  public List<UUID> getUserIds() {
+    List<UUID> userIds = new ArrayList();
+    
+    for(User user : users) {
```
Comment (erschwartz) [inline src/main/java/codeu/model/store/basic/UserStore.java]: Fix spacing here

File: src/main/java/codeu/model/store/basic/UserStore.java (hunk 4 above)
Comment (erschwartz) [reply src/main/java/codeu/model/store/basic/UserStore.java]: You missed this comment

**Review by erschwartz — CHANGES_REQUESTED** [review]

**Review by erschwartz — APPROVED** [review]
LGTM, thanks

**Pull request merged**